src/main/java/io/neonbee/internal/processor/EntityProcessor.java

```diff
@@ -88,148 +88,119 @@ public void init(OData odata, ServiceMetadata serviceMetadata) {
-            EdmEntitySet edmEntitySet, UriInfo uriInfo, ODataResponse response, ContentType responseFormat,
-            HttpStatusCode responseStatus) {
-        return asyncEntity -> handleResult(asyncEntity, processPromise -> {
-            // The result entity that will be returned
-            Entity foundEntity = null;
-
-            // The unfiltered list of entities provided for the requested entityset. At this point in time the
-            // provided key predicates was not yet applied.
-            List<Entity> receivedEntities = asyncEntity.result().getEntities();
-            if (receivedEntities.isEmpty()) {
-                response.setStatusCode(HttpStatusCode.NOT_FOUND.getStatusCode());
-                processPromise.complete();
-                return;
-            }
-
-            try {
-                foundEntity = findEntityByKeyPredicates(routingContext, uriResourceEntitySet, receivedEntities);
-            } catch (Exception e) {
-                LOGGER.correlateWith(routingContext).error(e.getMessage(), e);
-                processPromise
-                        .fail(new ODataApplicationException("Error during processing key predicates: " + e.getMessage(),
-                                HttpStatusCode.INTERNAL_SERVER_ERROR.getStatusCode(), Locale.ENGLISH));
-            }
-
-            EntitySerializerOptions.Builder optionsBuilder =
-                    EntitySerializerOptions.with().contextURL(ContextURL.with().entitySet(edmEntitySet).build());
-
-            // Apply $select system query option. The selection is handled by the olingo odata server lib,
-            // therefore only the context URL and serializer options have to be configured correctly.
-            SelectOption selectOption = uriInfo.getSelectOption();
-            if (selectOption != null) {
-                // Property names of the $select, in order to build the context URL
-                try {
-                    optionsBuilder.contextURL(ContextURL
-                            .with().entitySet(edmEntitySet).selectList(odata.createUriHelper()
-                                    .buildContextURLSelectList(edmEntitySet.getEntityType(), null, selectOption))
-                            .build());
-                    optionsBuilder.select(selectOption);
-                } catch (SerializerException e) {
-                    processPromise.fail(
-                            new ODataApplicationException("Error during processing select option: " + e.getMessage(),
-                                    HttpStatusCode.INTERNAL_SERVER_ERROR.getStatusCode(), Locale.ENGLISH));
-                }
-            }
-
-            // Return the OData response
+    private Handler<EntityWrapper> handleReadEntityResult(UriResourceEntitySet uriResourceEntitySet, UriInfo uriInfo,
+            ODataResponse response, ContentType responseFormat, HttpStatusCode responseStatus,
```

Comment:
This value is currently always HttpStatus.OK. Maybe it make sense to not pass it. It would reduce the signature by 1 line which is also nice. If we need to parse it in future then we can pass it but for now it does not make sense
`HttpStatusCode responseStatus,`